### DISCUSSION TITLE
Setting up rust-tools with lazyvim

### DISCUSSION THREAD

**Discussion by viralams** [LazyVim/LazyVim#302] [Q&A] [answered]
I know that rust-tools has a conflict with the `rust-analyzer` as mentioned [here](https://github.com/simrat39/rust-tools.nvim#setup). I am trying to fiddle with the settings in my lazy nvim plugins when the lsp server loads. I have not had any luck. Can anyone help with this setup?

**Comment by dpetka2001** [comment, marked as answer]
You could try [this discussion](https://github.com/LazyVim/LazyVim/discussions/42). It also has rust-tools configured. So maybe you could get some ideas to get it working.

**Reply by viralams** [reply]
I was able to get it working with the provided link. I was confused on setting up `on_attach` and now seems to be a breeze. Thanks for the help.

**Reply by geo-ant** [reply]
Could you elaborate on how you got on_attach to work? The config in the link above does not have a custom on_attach and it seems to be broken for me as well...

**Reply by dpetka2001** [reply]
Since that post was created, an extra for [Rust](https://www.lazyvim.org/extras/lang/rust) has been created to make it easier for users to setup a working environment. I don't use Rust, but maybe you could look at that.
